src/main/scala/com/softwaremill/realworld/users/UsersEndpoints.scala

```diff
@@ -0,0 +1,67 @@
+package com.softwaremill.realworld.users
+
+import com.softwaremill.realworld.db.{Db, DbConfig}
+import com.softwaremill.realworld.utils.*
+import io.getquill.SnakeCase
+import sttp.model.StatusCode
+import sttp.tapir.generic.auto.*
+import sttp.tapir.json.zio.jsonBody
+import sttp.tapir.server.ServerEndpoint.Full
+import sttp.tapir.ztapir.*
+import sttp.tapir.{EndpointInput, PublicEndpoint, Validator}
+import zio.json.{DeriveJsonDecoder, DeriveJsonEncoder}
+import zio.{Cause, Exit, ZIO, ZLayer}
+
+import javax.sql.DataSource
+
+class UsersEndpoints(usersService: UsersService, base: BaseEndpoints):
+
+  import UsersEndpoints.given
+
+  val get: ZServerEndpoint[Any, Any] = base.secureEndpoint.get
+    .in("api" / "user")
+    .out(jsonBody[User])
+    .serverLogic(session =>
```

Comment:
for ZIO, there's special integration using `import sttp.tapir.ztapir.*`, see: https://tapir.softwaremill.com/en/latest/server/ziohttp.html

Then, you can use `.zServerLogic` properly taking advantage of ZIO's error channel